### DISCUSSION TITLE
[CRUD] Unallowed State Transition Response Codes

### DISCUSSION THREAD

**Discussion by GioAc96** [mia-platform/community#208] [Q&A] [answered]
When requesting an unallowed state transition on a document on the CRUD service (ie: from `PUBLIC` to `DELETED`) the CRUD service responds with a `404 NOT FOUND` status code and the following response body:
```JSON
{
    "error": "not found"
}
```

However, I find this response not helpful, as it is the same that is returned when the document is not found. I believe that the appropriate response would be a `400 BAD REQUEST` with a response body along the lines of:
```JSON
{
  "error": "transitions from state 'PUBLIC' to 'DELETED' are not allowed"
}
```

**Comment by ThisIsDemetrio** [comment, marked as answer]
Hi @GioAc96 and thank you for your input.

You re right. It is indeed a non valid request (since [you cannot change the state of a document from `PUBLIC` to `DELETED` straight away](https://github.com/mia-platform/crud-service#document-state-management).

Since it looks more of an issue than a configuration/usage problem, I'd suggest you to open a issue in the [CRUD Service GitHub Issues page](https://github.com/mia-platform/crud-service/issues) so we can easily keep track of the problem and find a solution for a better error management.

Again, thank you for noticing that
|D|

**Reply by GioAc96** [reply]
Done! Here's a link to the issue I've created:

https://github.com/mia-platform/crud-service/issues/53
